ecosystem/sep-0045.md

```diff
@@ -0,0 +1,431 @@
+| `address`       | `C...` string | The **Client Account** address (`C...`) that the **Client** wishes to authenticate with the **Server**.                                                                                                                                                                                                                                                                                                                    |
+| `home_domain`   | string        | A **Home Domain**. Servers that generate tokens for multiple **Home Domain**s can use this parameter to identify which home domain the **Client** hopes to authenticate with.                                                                                                                                                                                                                                              |
+| `client_domain` | string        | (optional) a **Client Domain**. Supplied by **Clients** that intend to verify their domain in addition to the **Client Account**. See [Verifying the Client Domain](#verifying-the-client-domain). **Servers** should ignore this parameter if the **Server** does not support **Client Domain** verification, or the **Server** does not support verification for the specific **Client Domain** included in the request. |
+
+Example:
+
+```
+GET https://auth.example.com/?address=CCIBUCGPOHWMMMFPFTDWBSVHQRT4DIBJ7AD6BZJYDITBK2LCVBYW7HUQ&home_domain=example.com
+```
+
+#### Response
+
+##### Success
+
+On success the endpoint must return `200 OK` HTTP status code and a JSON object with the following fields:
+
+- `authorization_entries`: a list of XDR-encoded `SorobanAuthorizationEntry`. It contains an entry for each of the
+  **Client Account**'s signers in addition to a signed entry from the **Home Domain Address** and optionally the
+  **Client Domain Address**. The entries are not guaranteed to be in any specific order.
+
+Each entry's `root_invocation` function is a `contract_fn` with the following with no additional sub invocations:
+
+- `contract_address` is the `WEB_AUTH_CONTRACT_ID` from the **Server**'s `stellar.toml`
+- `function_name` is `web_auth_verify`
+- `args`:
+  - `address` matches the `address` from the request
+  - `home_domain` matches the `home_domain` from the request
+  - `home_domain_address` matches the **Home Domain Address**
+  - `web_auth_domain` matches the **Server**'s domain
+  - `client_domain` matches the **Client**'s domain
+  - `client_domain_address` matches the **Client Domain Address**
```

Comment:
The list of args here are stored in an array, and so are not self describing, and are difficult to insert or remove new values over time if needed to. One of the advantages of SEP-10 was that because data was stored in data operations, new fields could be added as needed.

I think there's value in considering the args be a single value, a map, and that map contains mappings of symbols to values.

Similar to SEP-10, it should state that the list is not exhaustive so that implementations do not assume it is, and so that it can be expanded in the future.

Reply:
This was the approach I went with initially but I discovered during implementation that the args map would not deserialize when it exceeded a certain size. Maybe there's a workaround or I implemented it incorrectly, I'll continue looking into this.

Reply:
This is very interesting. Do you happen to have an example of this in code I could experiment with?

Reply:
Yep. https://gist.github.com/philipliu/5524addbfe41abb48f5e292c8f3d2670

The only difference between the transactions is that the failing one has a nonce argument.

Reply:
When I simulate the transaction with the follow command, I get the following resources required:

```
$ stellar contract invoke \
    --id CCUFWWBPHQZGEHDQX76OJF32GMWGHK43A4L63HM6VV7NQ7ZIWCV4JLOM \
    --build-only \
    -- \
    web_auth_verify \
    --args '{"address":"CDYOQJLKZWHZ2CVN43EVEQNDLEN544IGCO5A52UG4YS6KDN5QQ2LUWKY","home_domain":"localhost:8080","home_domain_address":"GDJLBYYKMCXNVVNABOE66NYXQGIA5AC5D223Z2KF6ZEYK4UBCA7FKLTG","web_auth_domain":"localhost:8080"}' \
    | stellar tx simulate \
    | stellar xdr decode --type TransactionEnvelope \
    | jq -r '.tx.tx.ext.v1.resources | "| Instructions | Read Bytes | Write Bytes |\n|---|---|---|\n| \(.instructions) | \(.read_bytes) | \(.write_bytes) |"'
```
| Instructions | Read Bytes | Write Bytes |
|---|---|---|
| 1,791,370 | 3,564 | 148 |

Those limits don't look like they're anywhere near the limits posted here: https://developers.stellar.org/docs/networks/resource-limits-fees

What am I doing differently?

Reply:
We worked out offline that the issue is the map wasn't sorted in the failing tx.